weldx/core.py

```diff
@@ -324,7 +324,7 @@ def __eq__(self, other: Any) -> bool:
         if not isinstance(self.data, MathematicalExpression):
             if not isinstance(other.data, pint.Quantity):
                 return False
-            return self._data.identical(other.data_array)
+            return self._data.identical(other.data_array)  # type: ignore
```

Comment:
shouldn't mypy inspect the return type of the identical method then?

Reply:
The problem here is a bit different. `self._data` isn't guaranteed to be a xarray at this point, or at least, mypy doesn't see it that way. So it complains about something like "`Any` has no method `identical`". We could change the code to give this guarantee, but this might have side effects requiring more attention. If the code does what it is meant to do I acknowledge that there might be an issue by just adding `# type: ignore`. Not every reported issue is really relevant and needs to be taken care of.